public/app/features/correlations/CorrelationsPage.tsx

```diff
@@ -89,26 +88,38 @@ export default function CorrelationsPage() {
 
   useEffect(() => {
     if (!remove.error && !remove.loading && remove.value) {
-      fetchCorrelations();
+      fetchCorrelations({ page });
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [remove.error, remove.loading, remove.value, fetchCorrelations]);
 
+  useEffect(() => {
+    fetchCorrelations({ page });
+  }, [fetchCorrelations, page]);
```

Comment:
with the latest changes everything looks good, but when deleting the last item of a page, both of these effects will trigger, causing a duplicate fetch 

![Screenshot 2023-06-27 at 12 09 15](https://github.com/grafana/grafana/assets/1170767/e5966e7d-0676-40ed-93f0-1436b9484449)

Reply:
Good catch! try now :D